### ISSUE TITLE
Weird behaviour after doctrine-bundle update 2.0.6

### ISSUE THREAD

**Issue by acasademont** [php-pm/php-pm-httpkernel#161]
Just to keep track of it, seems like the new doctrine-bundle default behaviour interferes with our clear or reset strategy we have in the Symfony bootstrap.

https://github.com/doctrine/DoctrineBundle/issues/1114

**Comment by dnna** [comment]
That looks strange indeed. I'll try to have a look in the next few days, as this will probably affect my projects as well after upgrading. The PHP-PM code itself doesn't look wrong, it only resets if the entity manager is not open (e.g. an exception occurred that closed it), otherwise it only clears the existing one. Maybe doctrine bundle introduced their own reset logic using the `kernel.reset` tag and PHP-PM does not need to clear it anymore at all?

**Comment by dnna** [comment]
Btw, if DoctrineBundle now performs its own reset we should probably change remove our reset logic in PHP-PM's postHandle or make it conditional on `Doctrine\Bundle\DoctrineBundle\Registry` not having `ResetInterface`. Otherwise we are resetting twice now.

**Comment by acasademont** [comment]
yes, exactly, it's as easy as to check if the doctrine service is implementing the `ResetInterface` or not

**Comment by mathieudz** [comment]
I'm having memory leaks after upgrading from doctrine-bundle 2.0.2 to 2.0.6. Workers reach max memory after 500 requests while the workers are only restarted after 1000 requests. I've downgraded it back to 2.0.2

**Comment by acasademont** [comment]
Interesting side effect, I didn’t even notice. I’m going to check our
production logs. In any case, we too reverted to 2.0.2

**Comment by acasademont** [comment]
@mathieudz didn't see any memory limit crashes on our logs. Still, we should also keep an eye on it.
